src/CP/core/search/ilds.jl

```diff
@@ -7,30 +7,26 @@ initroot!(toCall::Stack{Function}, ::ILDSearch, model::CPModel, variableHeuristi
 generic function to instantiate the research based on a specific Strategy <: SearchStrategy. The max discrepancy correspond to the number of branchable variables 
 at the beginning of the search. Calls to expandIlds! with a decreasing discrepancy is stacked in the toCall Stack. 
 """
-function initroot!(toCall::Stack{Function}, ::ILDSearch , model::CPModel, variableHeuristic::AbstractVariableSelection, valueSelection::ValueSelection)
-    isboundedlist = [!isbound(v) for (k,v) in model.variables]
-    @assert !isempty(isboundedlist) "initialisation failed : no declared variables"
-    depth = sum(isboundedlist)
-    
+function initroot!(toCall::Stack{Function}, search::ILDSearch , model::CPModel, variableHeuristic::AbstractVariableSelection, valueSelection::ValueSelection)
+
     # Note that toCall stack is a LIFO data structure, expandIlds with a discrepancy threshold of 0 will be the first one to execute (then with 1, 2, 3, etc.)
-    for k in depth:-1:1
-        push!(toCall, (model) -> (restart_search!(model); expandIlds!(toCall,k,depth, nothing, model, variableHeuristic, valueSelection)))
+    for k in search.d:-1:1
+        push!(toCall, (model) -> (restart_search!(model); expandIlds!(toCall,k, model, variableHeuristic, valueSelection)))
```

Comment:
ILDSearch now uses the user-defined attribute `d` that defines the max discrepancy authorized during the search.